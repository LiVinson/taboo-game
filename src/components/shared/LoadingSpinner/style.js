import styled, { keyframes } from 'styled-components'

const spin = keyframes`
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }

`

export const StyledSpinner = styled.div`
	position: absolute;
	left: 50%;
	top: 50%;
	transform: translate(-50%, -50%);
	border: ${(props) => props.theme.color.grayDark1} solid 1.8rem;
	border-radius: 50%;
	border-top: ${(props) => props.theme.color.primary} solid 1.8rem;
	width: 9rem;
	height: 9rem;
	animation: ${spin} 2s linear infinite;
  	box-shadow: 0px 2px 1rem rgba(0, 0, 0, 0.5); 
`
