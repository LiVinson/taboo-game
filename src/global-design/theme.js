const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
