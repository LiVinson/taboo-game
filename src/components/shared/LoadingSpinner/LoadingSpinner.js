import React from 'react'
import { StyledSpinner } from './style'

const LoadingSpinner = () => {
	return <StyledSpinner></StyledSpinner>
}

export default LoadingSpinner
