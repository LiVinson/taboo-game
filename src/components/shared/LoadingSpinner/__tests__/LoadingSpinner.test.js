import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../../global-design/theme'
import LoadingSpinner from '../LoadingSpinner'

describe('LoadingSpinner', () => {
	test('renders correctly', () => {
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<LoadingSpinner />
				</ThemeProvider>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})
})
