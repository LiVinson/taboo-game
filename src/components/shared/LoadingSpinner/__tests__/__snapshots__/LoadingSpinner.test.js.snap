// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly 1`] = `
.c0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border: #757575 solid 1.8rem;
  border-radius: 50%;
  border-top: #1982C4 solid 1.8rem;
  width: 9rem;
  height: 9rem;
  -webkit-animation: hGPBYj 2s linear infinite;
  animation: hGPBYj 2s linear infinite;
  box-shadow: 0px 2px 1rem rgba(0,0,0,0.5);
}

<div
  className="c0"
/>
`;
